### PR TITLE
fix(dataset): Page blanks on large data load

### PR DESCRIPTION
**Pull request by nikolagigic**
### SUMMARY
Set the pagination to `true` to prevent chart page from crashing due to heavy data load.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:
No Pagination 2

After:
With Pagination 2

### TEST PLAN
1. Navigate to **Datasets** page
2. Select **wb_health_population**
3. Set the **Visualization Type** to **Linear Chart**
4. Set the **Time Range** to **No Filter**
5. Run the query
6. Under the **Data** panel select **View Samples** tab

Data should load in couple of seconds.

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Changes UI
- [ ] Requires DB Migration.
- [ ] Confirm DB Migration upgrade and downgrade tested.
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Comment by junlincc** [comment]
i'm still getting the blank page for some reasons.😞 @nikolagigic 
ezgif-4-c54fea65fbff

**Review by junlincc — COMMENTED** [review]
i'm still getting the blank page for some reasons.😞 @nikolagigic
also the data section took few clicks to open, im sure it's because of the large dataset, that's a separate issue
ezgif-4-c54fea65fbff

**Comment by rusackas** [comment]
@nikolagigic The error seems to be the `toString` trying to chew on a `null` in `DataTableControl.tsx`. We should fix that, and it might be good to add an `<ErrorBoundary>` somewhere in all of this so the whole page doesn't crash in the first place.

As for the fix, we can either check if the value is null, or (if it's useful) use `String.valueOf(value)` instead of `value.toString()` which would take the `null` and return the string `"null"`

File: superset-frontend/spec/javascripts/explore/utils_spec.jsx [hunk 1]
```diff
@@ -200,6 +201,13 @@ describe('exploreUtils', () => {
     });
   });
 
+  describe('getSamplesPageSize', () => {
+    it('generates samples table', () => {
+      const pageSize = getSamplesPageSize(500);
+      expect(pageSize).toEqual(20);
+    });
+  });
```
Comment (villebro) [inline superset-frontend/spec/javascripts/explore/utils_spec.jsx:209]: Could we add a few more typical and extreme cases, like 0, 1, 100 and 1000000?

File: superset-frontend/src/explore/exploreUtils.js [hunk 2]
```diff
@@ -250,6 +250,14 @@ export function postForm(url, payload, target = '_blank') {
   document.body.removeChild(hiddenForm);
 }
 
+export function getSamplesPageSize(columnsLength) {
```
Comment (villebro) [inline superset-frontend/src/explore/exploreUtils.js:253]: Nit: could this be `columnCount`?

File: superset-frontend/src/explore/components/DataTablesPane.tsx [hunk 3]
```diff
@@ -191,11 +196,13 @@ export const DataTablesPane = ({
       if (data[type]?.length === 0) {
         return <span>No data</span>;
       }
+      const renderWithPagination = true;
       return (
         <TableView
           columns={columns[type]}
           data={filteredData[type]}
-          withPagination={false}
+          withPagination={renderWithPagination}
```
Comment (villebro) [inline superset-frontend/src/explore/components/DataTablesPane.tsx:204]: Perhaps we don't need the `const` above for this, just do `true` here.

**Review by villebro — COMMENTED** [review]
A few first pass comments

**Comment by junlincc** [comment]
pulled the latest its still blanking from my end 😞

File: superset-frontend/src/explore/components/DataTablesPane.tsx (hunk 3 above)
Comment (nikolagigic) [reply superset-frontend/src/explore/components/DataTablesPane.tsx:204]: Lint test is going to throw: `Value must be omitted for boolean attributes`

File: superset-frontend/src/explore/exploreUtils.js [hunk 4]
```diff
@@ -248,6 +248,14 @@ export function postForm(url, payload, target = '_blank') {
   document.body.removeChild(hiddenForm);
 }
 
+export function columnCount(columnsLength) {
+  let pageSize;
+  if (columnsLength) {
+    pageSize = Math.ceil(Math.max(5, 10000 / columnsLength));
+  }
+  return pageSize || 50;
+}
```
Comment (villebro) [inline superset-frontend/src/explore/exploreUtils.js:257]: This function could be named more appropriately, e.g. `getDataTablePageSize` or similar.

File: superset-frontend/spec/javascripts/explore/utils_spec.jsx [hunk 5]
```diff
@@ -200,6 +201,25 @@ describe('exploreUtils', () => {
...
+    });
+    it('divides samples data into pages of 20', () => {
+      const pageSize = columnCount(1);
+      expect(pageSize).toEqual(10000);
+    });
+    it('divides samples data into pages of 20', () => {
+      const pageSize = columnCount(1000000);
+      expect(pageSize).toEqual(5);
+    });
+  });
```
Comment (villebro) [inline superset-frontend/spec/javascripts/explore/utils_spec.jsx:221]: This probably comes down to preference, but I think we could get by with a single `it` here, just doing multiple `expect(columnCount(xxx)).toEqual(yyy);` in sequence, as this test is pretty simple. Just make sure the test description is clear on what's being tested. But if we do keep separate `it`s here, make sure to update the descriptions appropriately (all have a reference to 20, which only applies to the first case).

Comment (nikolagigic) [reply superset-frontend/spec/javascripts/explore/utils_spec.jsx:221]: yep, forgot to change the description. Will move it under one `it`

File: superset-frontend/src/explore/components/DataTablesPane.tsx (hunk 3 above)
Comment (villebro) [reply superset-frontend/src/explore/components/DataTablesPane.tsx:203]: Right, it should just be `withPagination` if it's set to true.

**Comment by villebro** [comment]
Tested with both very wide (300+ cols) and narrow (8 cols), and it seems to work nicely on both. Just needs a rebase and those last changes and this should be good to go.

**Review by junlincc — APPROVED** [review]
ezgif-6-228cd2b652f8

this is AWESOME Nikola!! working properly finally, we can merge now if the code look alright

File: superset-frontend/src/explore/components/DataTablesPane.tsx [hunk 6]
```diff
@@ -191,11 +195,13 @@ export const DataTablesPane = ({
...
```
Comment (rusackas) [inline superset-frontend/src/explore/components/DataTablesPane.tsx:203]:
```suggestion
          withPagination
```
^ I think this is what @villebro is suggesting... does this still throw errors?

Comment (nikolagigic) [reply superset-frontend/src/explore/components/DataTablesPane.tsx]: yep, this one works fine. thanks!

**Review by villebro — APPROVED** [review]
LGTM

**Pull request merged**